src/esp/sim/Simulator.h

```diff
@@ -765,6 +773,13 @@ class Simulator {
   int navMeshVisPrimID_ = esp::ID_UNDEFINED;
   esp::scene::SceneNode* navMeshVisNode_ = nullptr;
 
+  /**
+   * @brief Tracks whether or not the simulator was initialized
+   * to load textures.  Because we cache mesh loading, this should
+   * *not* be changed without calling close() first
+   */
+  int requiresTextures_ = -1;
```

Comment:
Minor suggestion: you could use `Containers::Optional<bool>` for a cleaner "tri bool" state (unknown / true / false) :)

Reply:
I like it!